src/lib/parser.c

```diff
@@ -184,6 +184,19 @@ __Z_INLINE parser_error_t parser_printPublicKey(publickey_t *pk,
     return parser_ok;
 }
 
+__Z_INLINE parser_error_t parser_printSignature(raw_signature_t *s,
+                                                char *outVal, uint16_t outValLen,
+                                                uint8_t pageIdx, uint8_t *pageCount) {
+
+    // REVIEW
+    char outBuffer[128];
```

Comment:
if you expect the string to be 128 bytes, always add at least one more byte to the array for zero termination. More about this in the comment below.